src/tensorflow.cc

```diff
@@ -595,68 +598,20 @@ SetStringInputTensor(
   }
 #endif  // TRITON_ENABLE_GPU
 
-  // Parse content and assign to 'tensor'. Each string in 'content'
-  // is a 4-byte length followed by the string itself with no
-  // null-terminator.
-  while (content_byte_size >= sizeof(uint32_t)) {
-    if (element_idx >= request_element_cnt) {
-      RESPOND_AND_SET_NULL_IF_ERROR(
-          response,
-          TRITONSERVER_ErrorNew(
-              TRITONSERVER_ERROR_INVALID_ARG,
-              std::string(
-                  "unexpected number of string elements " +
-                  std::to_string(element_idx + 1) + " for inference input '" +
-                  name + "', expecting " + std::to_string(request_element_cnt))
-                  .c_str()));
-      FillStringTensor(
-          tensor, tensor_offset + element_idx,
-          request_element_cnt - element_idx);
-      free(contiguous_buffer);
-      return cuda_copy;
-    }
-
-    const uint32_t len = *(reinterpret_cast<const uint32_t*>(content));
-    content += sizeof(uint32_t);
-    content_byte_size -= sizeof(uint32_t);
-
-    if (content_byte_size < len) {
-      RESPOND_AND_SET_NULL_IF_ERROR(
-          response,
-          TRITONSERVER_ErrorNew(
-              TRITONSERVER_ERROR_INVALID_ARG,
-              std::string(
-                  "incomplete string data for inference input '" +
-                  std::string(name) + "', expecting string of length " +
-                  std::to_string(len) + " but only " +
-                  std::to_string(content_byte_size) + " bytes available")
-                  .c_str()));
-      FillStringTensor(
-          tensor, tensor_offset + element_idx,
-          request_element_cnt - element_idx);
-      free(contiguous_buffer);
-      return cuda_copy;
-    }
+  std::vector<std::pair<const char*, const uint32_t>> str_list;
+  err = ValidateStringBuffer(
+      content, content_byte_size, request_element_cnt, name, &str_list);
+  // Set string values.
+  for (; element_idx < str_list.size(); ++element_idx) {
+    const auto& [addr, len] = str_list[element_idx];
+    TRITONTF_TensorSetString(tensor, tensor_offset + element_idx, addr, len);
+  }
 
-    TRITONTF_TensorSetString(tensor, tensor_offset + element_idx, content, len);
-    content += len;
-    content_byte_size -= len;
-    element_idx++;
-  }
-
-  if ((*response != nullptr) && (element_idx != request_element_cnt)) {
```

Comment:
`(element_idx != request_element_cnt)`

Don't we need this check? I think with the current changes it should look something like below:

```
request_element_cnt != str_list.size()
```

Reply:
I see.. The check is part of ValidateStringBuffer.